Framework/DataObjects/src/EventWorkspaceMRU.cpp

```diff
@@ -139,18 +139,22 @@ void EventWorkspaceMRU::insertE(size_t thread_num, EType data,
  * @param index :: index to delete.
  */
 void EventWorkspaceMRU::deleteIndex(const EventList *index) {
-  {
+  // Avoid locks when the buffers are empty
+  // Can significantly speed up EventList::mutableHistogramRef()
+  if (!m_bufferedDataE.empty()) {
```

Comment:
Another thread might be writing to `m_bufferedDataE` at the same time, so I am not sure why we can do this outside the lock?